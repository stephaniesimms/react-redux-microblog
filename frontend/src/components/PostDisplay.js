import React, { Component } from 'react';
import './PostDisplay.css';

/** Display a post:
 *
 * - show edit/delete buttons (& call parent on action)
 * - show vote count and +/- buttons (& call parent on action)
 *
 */
class PostDisplay extends Component {
  constructor(props) {
    super(props);
    this.doVoteUp = this.doVoteUp.bind(this);
    this.doVoteDown = this.doVoteDown.bind(this);
  }

  doVoteUp() {
    this.props.doVote('up');
  }

  doVoteDown() {
    this.props.doVote('down');
  }

  render() {
    const { deletePost, post, showEditForm } = this.props;

    return (
      <div className='PostDisplay'>
        <h3>{post.title}</h3>
        <p><em>{post.description}</em></p>
        <p className='mb-4'>{post.body}</p>

        <div className='PostDisplay-right'>
          <i className='fa fa-edit text-info'
            onClick={showEditForm} />

          <i className='fa fa-times text-danger'
            onClick={deletePost} />

          <div className='PostDisplay-votes'>
            <b>votes: {post.votes}</b>

            <i className='fas fa-grin-stars text-warning'
              onClick={this.doVoteUp} />
            <i className='fas fa-dizzy text-danger'
              onClick={this.doVoteDown} />
          </div>
        </div>
      </div>
    );
  }
}

export default PostDisplay;