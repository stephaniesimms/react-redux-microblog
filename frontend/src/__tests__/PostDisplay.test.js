import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostDisplay from '../components/PostDisplay';

it('renders without crashing', function () {
  shallow(<PostDisplay post="test" />);
});

it('matches snapshot', function () {
  let wrapper = shallow(<PostDisplay post="test"/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});