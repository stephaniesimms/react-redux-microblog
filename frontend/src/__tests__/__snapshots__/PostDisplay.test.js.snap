// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="PostDisplay"
>
  <h3 />
  <p>
    <em />
  </p>
  <p
    className="mb-4"
  />
  <div
    className="PostDisplay-right"
  >
    <i
      className="fa fa-edit text-info"
    />
    <i
      className="fa fa-times text-danger"
    />
    <div
      className="PostDisplay-votes"
    >
      <b>
        votes: 
      </b>
      <i
        className="fas fa-grin-stars text-warning"
        onClick={[Function]}
      />
      <i
        className="fas fa-dizzy text-danger"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
